apps/discordbot/bot.py

```diff
@@ -125,31 +129,92 @@ def check_thumb(react, _):
 def emoji_letters(chars):
     return [unicodedata.lookup(f"REGIONAL INDICATOR SYMBOL LETTER {c}") for c in chars]
 
-def get_xkcd_soup_info(comic_soup):
+
+def is_valid_xkcd_command(msg):
+    """ Returns whether msg contains an valid request. """
+
+    arguments = msg.split()
+    if len(arguments) > 3 or len(arguments) < 2:
+        return False
+    elif arguments[1] in VALID_XKCD_COMMANDS:
+        if arguments[1] == VALID_XKCD_COMMANDS[4] or arguments[1] == VALID_XKCD_COMMANDS[5]:
+            return len(arguments) == 3 and arguments[2].isdigit()
+        else:
+            return len(arguments) == 2
+    else:
+        return False
+
+
+async def get_xkcd(message):
+    """
+    Displays either a random XKCD comic, a specific issue, or the '!xkcd' help panel.
+    Assumes that a valid command is contained within msg (checked by is_valid_xkcd_command(String)).
+    """
+    msg = message.content.lower().split();
+    cmd = msg[1]
+    if cmd == VALID_XKCD_COMMANDS[0] or cmd == VALID_XKCD_COMMANDS[1]:
+        await get_random_xkcd(message)
+    elif cmd == VALID_XKCD_COMMANDS[2] or cmd == VALID_XKCD_COMMANDS[3]:
+        await message.channel.send(XKCD_HELP_MSG)
+    elif cmd == VALID_XKCD_COMMANDS[4] or cmd == VALID_XKCD_COMMANDS[5]:
+        await get_xkcd_issue(message, msg[2])
```

Comment:
This is hard to read. Just use the string literals or structure `VALID_XKCD_COMMANDS` differently